### PR TITLE
Add support for dependency (validation, completion) on workspace proj…

### PR DESCRIPTION
**Pull request by vrubezhny**
…ects #77

Add Hovers  for Workspace project artifacts

Signed-off-by: Victor Rubezhny <vrubezhny@redhat.com>

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java [hunk 1]
```diff
@@ -112,6 +117,57 @@ public Hover onTag(IHoverRequest request) throws Exception {
 		return null;
 	}
 
+	private Hover collectWorkspaceArtifactDescription(IHoverRequest request) {
```
Comment (mickaelistria) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java:120]: As this is supposed to return only one value and is not alerating the input parameter, we'd rather call it `getWorkspaceArtifactDescription` or `computeWorkspaceArtifactDescription` or `findWorkspaceArtifactDescription`.

**Review by mickaelistria — COMMENTED** [review]
This looks good. See just a few minor comments inline.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java [hunk 2]
```diff
@@ -112,6 +117,57 @@ public Hover onTag(IHoverRequest request) throws Exception {
...
+				.sorted(Comparator.comparing((MavenProject p) -> new DefaultArtifactVersion(p.getVersion())).reversed())
+			.	findFirst().map(p -> {
+					File file = p.getFile();
+					if (file == null && p != p.getExecutionProject()) {
+						file = p.getExecutionProject().getFile();
+					}
+					return file;
+				})
+				.map(f -> {
+					Model model = (Model)builder
```
[the remark this answers is not in the document]
Comment (vrubezhny) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java:142]: This was actually copied from `collectArtifactDescription()`. 
I'll remove this rebuilding.

File: lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java (hunk 1 above)
Comment (vrubezhny) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenHoverParticipant.java:120]: OK

**Comment by mickaelistria** [comment]
Thanks!

**Pull request merged**